src/interpreter/fvm.rs

```diff
@@ -218,24 +227,47 @@ impl<DB: Blockstore> Consensus for ForestExternsV2<DB> {
                 // check blocks are properly signed by their respective miner
                 // note we do not need to check extra's: it is a parent to block b
                 // which itself is signed, so it was willingly included by the miner
-                if let Ok(gas_used) = self.verify_block_signature(&bh_1) {
-                    total_gas += gas_used;
-                    if let Ok(gas_used) = self.verify_block_signature(&bh_2) {
+                let res = self.verify_block_signature(&bh_1);
```

Comment:
`verify_block_signature` has three possible outputs: fatal error, non-fatal error, ok(gas_used). We can handle these three branches without any nesting. Not a big deal, but it would make the code easier to read for me.

Reply:
I was juggling with possible options to make it more readable, the trick here is that a non-fatal error still requires the gas to be passed around so the eloquence suffers there. Let me see what I can do here.

Reply:
Fixed.